tests/datasets/test_streaming.py

```diff
@@ -34,17 +34,22 @@ def get_fake_samples_decoders(num_samples: int) -> Tuple[List[Dict[str, bytes]],
     return samples, decoders
 
 
-def write_synthetic_streaming_dataset(dirname: str, samples: List[Dict[str, bytes]], shard_size_limit: int) -> None:
+def write_synthetic_streaming_dataset(dirname: str, samples: List[Dict[str, bytes]], shard_size_limit: int,
+                                      compression: Optional[str]) -> None:
     first_sample_fields = list(samples[0].keys())
-    with StreamingDatasetWriter(dirname=dirname, fields=first_sample_fields,
-                                shard_size_limit=shard_size_limit) as writer:
+    with StreamingDatasetWriter(dirname=dirname,
+                                fields=first_sample_fields,
+                                shard_size_limit=shard_size_limit,
+                                compression=compression) as writer:
         writer.write_samples(samples=samples)
 
 
 @pytest.mark.timeout(10)
 @pytest.mark.parametrize('num_samples', [100, 10000])
 @pytest.mark.parametrize('shard_size_limit', [1 << 8, 1 << 16, 1 << 24])
-def test_writer(remote_local: Tuple[str, str], num_samples: int, shard_size_limit: int) -> None:
+@pytest.mark.parametrize('compression', [None, 'gz:5', 'gz'])
```

Comment:
Could this be its own test? Adding the parameterization here increases the number of tess across {num_samples} * {shard_size_limit} * {compression}; I don't believe that compression depends on the num samples are shard size limit?